File: digitalMenu/backend/src/main/java/com/digitalmenu/backend/business/mapper/UserMapper.java
package com.digitalmenu.backend.business.mapper;

import com.digitalmenu.backend.business.DTOs.usersDTOS.UserRequestDTO;
import com.digitalmenu.backend.business.DTOs.usersDTOS.UserResponseDTO;
import com.digitalmenu.backend.business.model.User;
import org.springframework.stereotype.Component;

@Component
public class UserMapper {
    public User toEntity(UserRequestDTO dto) {
        if (dto == null) return null;

        return User.builder().name(dto.getName())
                .email(dto.getEmail())
                .password(dto.getPassword())
                .build();
    }

    public UserResponseDTO toResponseDTO(User entity) {
        if (entity == null) return null;

        return UserResponseDTO.builder()
                .id(entity.getId())
                .name(entity.getName())
                .email(entity.getEmail())
                .createAt(entity.getCreatedAt())
                .build();
    }
}

File: digitalMenu/backend/src/main/kotlin/com/josefilipe/digitalmenu/backend/mapper/UserMapper.kt
package com.josefilipe.digitalmenu.backend.mapper

import com.josefilipe.digitalmenu.backend.dto.UserRequestDTO
import com.josefilipe.digitalmenu.backend.dto.UserResponseDTO
import com.josefilipe.digitalmenu.backend.model.User


fun UserRequestDTO.toEntity() = User(
    name = this.name,
    email = this.email,
    password = this.password,
    type = this.type
)


fun User.toResponseDTO() = UserResponseDTO(
    id = this.id,
    name = this.name,
    email = this.email,
    type = this.type
)